src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,210 @@
+   * alpha is from the GRT paper.
+   */
+  private final double alpha = 0.7;
+
+  /**
+   * Hyper-parameter for decreasing weights of methods between updates to coverage information. The
+   * name p is from the GRT paper.
+   */
+  private final double p = 0.5;
+
+  /** How often to recompute branch coverage, in steps. */
+  private final int branchCoverageInterval = 100;
+
+  /** Maximum number of successful calls so far to any method under test. */
+  private int maxSuccessfulCalls = 0;
+
+  /**
+   * The number of times {@code step()} has been invoked in {@link ForwardGenerator} to construct a
+   * new sequence.
+   */
+  private int stepNum = 0;
+
+  /**
+   * Construct a new instance of Bloodhound, which will choose from among the given operations.
+   *
+   * @param operations list of operations to copy
+   */
+  public Bloodhound(List<TypedOperation> operations) {
+    this.operationSimpleList = new SimpleArrayList<>(operations);
+  }
+
+  /**
+   * The branch coverage information for all methods under test is updated at every {@code
+   * branchCoverageInterval}'th call of this method.
+   */
+  private void updateBranchCoverageMaybe() {
+    stepNum += 1;
+    if (stepNum % branchCoverageInterval == 0) {
+      methodSelections.clear();
+      CoverageTracker.instance.collect();
+    }
+  }
+
+  /**
+   * Recompute weights for all methods under test. Each method under test is assigned a weight based
+   * on a weighted combination of the number of branches uncovered and the ratio between the number
+   * of times this method has been invoked and the maximum number of times any method under test has
+   * been invoked. The weighting scheme is based on Bloodhound in the Guided Random Testing (GRT)
+   * paper.
+   */
+  private void updateWeightsForOperations() {
+    // The number of methods under test, corresponds to |M| in the GRT paper.
+    int numOperations = this.operationSimpleList.size();
+
+    // Default weight is uniform probability.  If is used if no coverage details are available.
+    // This is the case for classes like java.lang.Object (not explicitly under test).
+    double weight = 1.0 / numOperations;
+
+    // Recompute weights for all operations.
+    for (TypedOperation operation : operationSimpleList) {
+      CoverageTracker.BranchCoverage covDet =
+          CoverageTracker.instance.getDetailsForMethod(operation.getName());
+
+      if (covDet != null) {
+        // Coverage details are available.
+
+        // The number of successful invocations of this method. Corresponds to succ(m).
+        Integer numSuccessfulInvocations = methodSuccessfulCalls.get(operation);
+        if (numSuccessfulInvocations == null) {
+          numSuccessfulInvocations = 0;
+        }
+
+        // If this method has no uncovered branches, or it is the most-invoked method,
+        // use the default weight and skip this step.
+        if (covDet.uncoveredBranches != 0 && numSuccessfulInvocations != maxSuccessfulCalls) {
```

Comment:
"maxSuccessfulCalls" and "numSuccessfulInvocations" are counting the same thing.  Please use consistent terminology (not "...Calls" vs "...Invocations").

Please make any other terminology consistent throughout the code, too.